scripts/Diffraction/isis_powder/osiris_routines/osiris_algs.py

```diff
@@ -237,9 +246,27 @@ def _correct_drange_overlap(merged_ws, drange_sets):
     return merged_ws
 
 
-def _merge_dspacing_runs(run_number, drange_sets, ws_group):
-    merged = MergeRuns(InputWorkspaces=ws_group, OutputWorkspace="OSIRIS" + run_number + WORKSPACE_SUFFIX.MERGED)
-    return _correct_drange_overlap(merged, drange_sets)
+def merge_dspacing_runs(focussed_runs, drange_sets, run_number):
+    if len(focussed_runs) == 1:
+        return [focussed_runs[0]]
+
+    extracted_spectras = [common.extract_ws_spectra(ws) for ws in focussed_runs]
+
+    have_same_spectras_count = len({len(spectras) for spectras in extracted_spectras}) == 1
+    if not have_same_spectras_count:
+        logger.warning("Cannot merge focussed workspaces with different number of spectras")
+        return [ws for ws in focussed_runs]
+
+    # Group workspaces located at the same index
+    transposed_spectras = list(map(list, zip(*extracted_spectras)))
+    # Merge workspaces located at the same index
+    merged_spectras = [
+        MergeRuns(InputWorkspaces=spectras, OutputWorkspace="OSIRIS" + run_number + WORKSPACE_SUFFIX.MERGED)
+        for spectras in transposed_spectras
+    ]
+    grouped_spectras = GroupWorkspaces(InputWorkspaces=merged_spectras)
```

Comment:
I'm getting a crash from this call to GroupWorkspaces. Please can you check if you see the same? In the PyCharm debugger the `merged_spectras` list looks a bit strange:
![image](https://user-images.githubusercontent.com/56295817/221547393-78657237-41a6-4e44-a97a-696c66fae4a4.png)
Does the OutputWorkspace name need to be different for each set of matched spectra in the list comprehension on L263? I'm wondering if the different iterations are overwriting the output of each other?

Reply:
Yes, this was the problem